langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/Round.java

```diff
@@ -31,7 +34,46 @@
 //)
 public class Round {
 
-    public static double round(double x) {
-        return Math.rint(x);
+    public static double round(double x, long fractionDigits) {
+        if (isSpecialValue(x)) {
+            return x;
+        }
+        if (fractionDigits == 0) {
+            return Math.rint(x);
+        }
+        if (fractionDigits > Integer.MAX_VALUE) {
+            // As per IEEE, exponent of double value lies from -308 to 307.
+            // Also, the maximum decimal digits that can have in double value are 15 or 16.
+            // Therefore, if `fractionDigits` is very large, `x` will not be changed.
+            return x;
+        }
+        if (fractionDigits < Integer.MIN_VALUE) {
```

Comment:
Can fractalDigits be less than 0?

Reply:
Yes please check https://github.com/ballerina-platform/ballerina-spec/issues/1074